drivers/gpio/gpio_max149x6.h

```diff
@@ -0,0 +1,143 @@
+	 *					industrial-digital-output-digital-input.html
+	 */
+
+	for (i = (encode) ? 0 : 2; i < 8; i++) {
+		data_bit = (data[0] >> (7 - i)) & 0x01;
+		result_bit = (crc5_result & 0x10) >> 4;
+		if (data_bit ^ result_bit)
+			crc5_result = crc5_poly ^ ((crc5_result << 1) & 0x1f);
+		else
+			crc5_result = (crc5_result << 1) & 0x1f;
+	}
+
+	for (i = 0; i < 8; i++) {
+		data_bit = (data[1] >> (7 - i)) & 0x01;
+		result_bit = (crc5_result & 0x10) >> 4;
+		if (data_bit ^ result_bit)
+			crc5_result = crc5_poly ^ ((crc5_result << 1) & 0x1f);
+		else
+			crc5_result = (crc5_result << 1) & 0x1f;
+	}
+
+	for (i = 0; i < 3; i++) {
+		data_bit = (extra_byte >> (7 - i)) & 0x01;
+		result_bit = (crc5_result & 0x10) >> 4;
+		if (data_bit ^ result_bit)
+			crc5_result = crc5_poly ^ ((crc5_result << 1) & 0x1f);
+		else
+			crc5_result = (crc5_result << 1) & 0x1f;
+	}
+
+	return crc5_result;
+}
+
+/*
+ * @brief Register write function for MAX14906
+ *
+ * @param dev - MAX14906 device config.
+ * @param addr - Register value to which data is written.
+ * @param val - Value which is to be written to requested register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int max149x6_reg_transceive(const struct device *dev, uint8_t addr, uint8_t val,
+							uint8_t *rx_diag_buff, uint8_t rw)
+{
+	uint8_t crc;
+	int ret;
+
+	uint8_t local_rx_buff[MAX149x6_MAX_PKT_SIZE] = {0};
+	uint8_t local_tx_buff[MAX149x6_MAX_PKT_SIZE] = {0};
+
+	const struct max149x6_config *config = dev->config;
+
+
+	struct spi_buf tx_buf = {
+				.buf = &local_tx_buff,
+				.len = config->pkt_size,
+				};
+	const struct spi_buf_set tx = {.buffers = &tx_buf, .count = 1};
+
+	struct spi_buf rx_buf = {
+				.buf = &local_rx_buff,
+				.len = config->pkt_size,
+				};
+	const struct spi_buf_set rx = {.buffers = &rx_buf, .count = 1};
+
+	if (config->crc_en & 0) {
+		rx_buf.len++;
+	}
+
+	local_tx_buff[0] = FIELD_PREP(MAX149x6_ADDR_MASK, addr) |
+				FIELD_PREP(MAX149x6_CHIP_ADDR_MASK, config->spi_addr) |
+				FIELD_PREP(MAX149x6_RW_MASK, rw & 0x1);
+	local_tx_buff[1] = val;
+
+	/* If CRC enabled calculate it */
+	if (config->crc_en) {
+		local_tx_buff[2] = max149x6_crc(&local_tx_buff[0], true);
+	}
+
+	/* write cmd & read resp at once */
+	ret = spi_transceive_dt(&config->spi, &tx, &rx);
+
+	if (ret) {
+		LOG_ERR("Err spi_transcieve_dt  [%d]\n", ret);
+		return ret;
+	}
+
+	/* if CRC enabled check readed */
+	if (config->crc_en) {
+		crc = max149x6_crc(&local_rx_buff[0], false);
+		if (crc != (local_rx_buff[2] & 0x1F)) {
+			LOG_ERR("READ CRC ERR (%d)-(%d)\n", crc, (local_rx_buff[2] & 0x1F));
+			return -EINVAL;
+		}
+	}
+
+	if (rx_diag_buff != NULL)
+		rx_diag_buff[0] = local_rx_buff[0];
+
+	/* In case of write we are getting 2 diagnostic bytes - byte0 & byte1
+	 * and pass them to diag buffer to be parsed in next stage
+	 */
+	if ((MAX149x6_WRITE == rw) && (rx_diag_buff != NULL))
+		rx_diag_buff[1] = local_rx_buff[1];
+	else
+		ret = local_rx_buff[1];
```

Comment:
Add braces to every `if`, `else`, `do`, `while`, `for` and `switch` body, even for single-line code blocks
https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style

Reply:
Thanks , solved on all places